### DISCUSSION TITLE
依赖版本问题

### DISCUSSION THREAD

**Discussion by BanKnight** [xmake-io/xmake#2220] [Q&A] [answered]
我自己参考官方的skia的写法，修改了package种对freetype的依赖版本设定
```
        add_deps("fontconfig", "freetype >=2.10<2.11")
```
但是跑起来的时候提示冲突
```
error: package(freetype/>=2.10<2.11:{}): conflict dependences with package(freetype/>=2.9:{})!
```

检查发现，是因为另一个依赖fontconfig的问题
skia=>fontconfig=>freetype/>=2.9
skia=>freetype=>/>=2.10<2.11

我理解2.10.X版本应该是完全符合才对，不知道为何会当作冲突

**Comment by xq114** [comment]
直接写版本 "freetype 2.10.x" 更好；如果要多个条件，条件之间加空格 "freetype >=2.10 <2.11"

**Reply by BanKnight** [reply]
试了一下，发现两种方法都不行。
```
error: package(freetype/2.10.4:{}): conflict dependences with package(freetype/>=2.9:{})!
```

**Comment by waruqi** [comment, marked as answer]
依赖了两个不同的版本，当然会冲突，要么走 `add_requireconfs("**.freetype", {override = true, version = "xxxx"})` 改写所有freetype依赖版本，保持一致

**Reply by xq114** [reply]
这是package里面的，add_deps，没有add_requireconf

**Reply by waruqi** [reply]
他没必要自己去改 skia package 里面的 add_deps ，直接用的时候 外面配置改就行了。。

**Comment by BanKnight** [comment]
add_requireconfs("freetype", {override = true, version = "2.10.4"})
冲突提示的确没有了。

不过从版本的写法语义上来说，是存在交集的，例如文中的2.10.4版本就完全符合。
结果就不太符合直觉。

**Reply by waruqi** [reply]
得用 

```lua
add_requireconfs("**.freetype", {override = true, version = "2.10.4"})
```

或者 

```lua
add_requireconfs("skia.freetype", {override = true, version = "2.10.4"})
add_requireconfs("skia.**.freetype", {override = true, version = "2.10.4"})
```

去改写全部依赖，或者是 skia 的全部依赖。。。

**Reply by BanKnight** [reply]
发现以上3种写法都是相同的报错
```lua
error: package(freetype/>=2.10:{}): conflict dependences with package(freetype/2.10.4:{})!
```
